Nautilus/Assets/Gadgets/UpgradeModuleGadget.cs

```diff
@@ -0,0 +1,114 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Diagnostics.CodeAnalysis;
+using Nautilus.Handlers;
+using Nautilus.Utility;
+
+namespace Nautilus.Assets.Gadgets;
+
+/// <summary>
+/// Represents a vehicle module (or upgrade) gadget
+/// </summary>
+public class UpgradeModuleGadget : Gadget
```

Comment:
Perhaps rename to "VehicleUpgradeModuleGadget", since the game considers "Vehicles" to be everything except the Cyclops